saleor/webhook/transport/asynchronous/transport.py

```diff
@@ -282,32 +307,69 @@ def trigger_webhooks_async_for_multiple_objects(
                             event_payload=payload,
                             event_type=event_type,
                         )
                     )
 
     if subscription_webhooks:
-        deliveries.extend(
-            create_deliveries_for_multiple_subscription_objects(
-                event_type=event_type,
-                subscribable_objects=[
-                    webhook_payload_data.subscribable_object
-                    for webhook_payload_data in webhook_payloads_data
-                ],
-                webhooks=subscription_webhooks,
-                requestor=requestor,
-                allow_replica=allow_replica,
-                pre_save_payloads=pre_save_payloads,
-                request_time=request_time,
+        subscribable_objects = [
+            webhook_payload_data.subscribable_object
+            for webhook_payload_data in webhook_payloads_data
+        ]
+        if is_deferred_payload:
+            _deferred_deliveries_to_create = []
+            for subscribable_object in subscribable_objects:
+                for webhook in subscription_webhooks:
+                    delivery = EventDelivery(
+                        status=EventDeliveryStatus.PENDING,
+                        event_type=event_type,
+                        webhook=webhook,
+                    )
+                    _deferred_deliveries_to_create.append(delivery)
+                    deferred_payload_data = prepare_deferred_payload_data(
+                        subscribable_object, requestor, request_time
+                    )
+                    deferred_deliveries_per_object[subscribable_object.pk].append(
+                        (delivery, deferred_payload_data)
+                    )
+
+            EventDelivery.objects.bulk_create(_deferred_deliveries_to_create)
+        else:
+            deliveries.extend(
+                create_deliveries_for_multiple_subscription_objects(
+                    event_type=event_type,
+                    subscribable_objects=subscribable_objects,
+                    webhooks=subscription_webhooks,
+                    requestor=requestor,
+                    allow_replica=allow_replica,
+                    pre_save_payloads=pre_save_payloads,
+                    request_time=request_time,
+                )
             )
-        )
+
+    if deferred_deliveries_per_object:
+        for _, deferred_deliveries in deferred_deliveries_per_object.items():
+            event_delivery_ids = [delivery.id for delivery, _ in deferred_deliveries]
+
+            # Trigger deferred payload generation task for each subscribable object.
+            generate_deferred_payloads.apply_async(
+                kwargs={
+                    "event_delivery_ids": event_delivery_ids,
+                    "deferred_payload_data": asdict(deferred_payload_data),
+                    "queue": queue,
+                },
+                queue=queue or settings.WEBHOOK_CELERY_QUEUE_NAME,
```

Comment:
This means: "use the same celery queue as the one for sending the webhooks". 
And IMO this is wrong as it means, that single queue will be shared between sending webhooks & building the webhook payloads. Few downsizes of using the same queue:
- in case of high traffic and a huge amount of webhooks to send in the queue, the building payload could be delayed, which means that the body could be changed, or object could not exist anymore. 
- sending webhooks could be delayed as right now the same queue also builds the payload. So assuming that you have checkout with sync webhooks to call, this could take a few seconds (max 20 secs) to process single task. It means, that this could downgrade the maximal number of webhooks that we're able to send by `WEBHOOK_CELERY_QUEUE_NAME`

IMHO, generating payloads should have its own queue, as we want to build a payload as fast as possible. Or at least using the default one, but using the same as the one for sending webhooks could cause a problem in case of huge amount of webhooks on queue.

Reply:
>in case of high traffic and a huge amount of webhooks to send in the queue, the building payload could be delayed, which means that the body could be changed, or object could not exist anymore.

That should not be an issue as there is similarly little value in delivering long-obsolete data.

>sending webhooks could be delayed as right now the same queue also builds the payload. So assuming that you have checkout with sync webhooks to call, this could take a few seconds (max 20 secs) to process single task.

That is a valid concern, generating events would happen at the expense of our webhook delivery capacity. We could use the default queue for payload generation until we implement the delivery fairness improvements we're working on.

Reply:
I also agree to use the default queue for this task. 
I don't see any more objections.

Reply:
Change to use default queue for payload generation.